File: easy/c/0083.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */


struct ListNode* deleteDuplicates(struct ListNode* head){
    if(head == NULL)return head;
    struct ListNode* ptr = head;
    struct ListNode* cur = head->next;
    while(head->next != NULL){
        if(head->next->val == head->val){
            cur = head->next;
            head->next = cur->next;
            free(cur);
        }else{
            head = head->next;
        }
    }
    return ptr;
}

File: easy/python/0083.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def deleteDuplicates(self, head: 'ListNode') -> 'ListNode':
        ptr = head
        if not head:
            return head
        prev = head.val
        while head.next:
            next_ = head.next
            while next_.val == prev :
                next_ = next_.next
                if not next_:
                    head.next = None
                    return ptr
            head.next = next_
            prev = next_.val
            head = next_
        return ptr